Update reference docs - remove metrics items

ericvn opened the pull request:
Please provide a description for what this PR is for.

This is a start of getting the update_ref_docs working with *master*.
Since mixer was already removed, some additional references needed to be removed from the script that was pulling in adapters from other repos.

Some work remains with removing some additional deprecated `mixer` pages, but that's outside the work in this PR.

istio-testing commented:
@ericvn: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 4bcd941dd86363138d1a50acb3701c1fbc658659 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/lint_istio.io/8268) | `/test lint_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ericvn: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 4bcd941dd86363138d1a50acb3701c1fbc658659 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/lint_istio.io/8268) | `/test lint_istio.io`
gencheck_istio.io | b4f6e140d83c9e96f1aa33668b06f2dca51b79e7 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/gencheck_istio.io/2412) | `/test gencheck_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ericvn: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio.io | b4f6e140d83c9e96f1aa33668b06f2dca51b79e7 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/gencheck_istio.io/2412) | `/test gencheck_istio.io`
lint_istio.io | b4f6e140d83c9e96f1aa33668b06f2dca51b79e7 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/lint_istio.io/8269) | `/test lint_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ericvn: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | b4f6e140d83c9e96f1aa33668b06f2dca51b79e7 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/lint_istio.io/8269) | `/test lint_istio.io`
gencheck_istio.io | 4269c2a83506709fb905f45b6ef128504a93c916 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/gencheck_istio.io/2413) | `/test gencheck_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ericvn: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio.io | 4269c2a83506709fb905f45b6ef128504a93c916 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/gencheck_istio.io/2413) | `/test gencheck_istio.io`
lint_istio.io | 4269c2a83506709fb905f45b6ef128504a93c916 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/lint_istio.io/8270) | `/test lint_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ericvn: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 17cc89766aab6b03136f1aeae50fb8f7a2dd104d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/8061/lint_istio.io/8271) | `/test lint_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ericvn commented:
I did end up deleting 2 files:
- docs/tasks/policy-enforcement/denial-and-list/index.md
- docs/tasks/policy-enforcement/rate-limiting/index.md
that were marked as deprecated. I'm not sure what they are being replaced with so maybe we need a bigger TODO for that.
Note that we do point to rate limiting in a few places like `Getting Started` and I removed that as well.

I also removed some features from the feature list. Three were deprecated, but two weren't:
- Statsd Integration | Stable
- Stackdriver Integration | Beta

ericvn commented:
@howardjohn @douglas-reid Would you look at this and make sure we aren't doing something excessive removing these pages from the istio.io docs. I assume the rate-limiting page would get updated with whatever comes from https://github.com/istio/istio/issues/25725.

douglas-reid commented:
fwiw, I think we can remove `Statsd Integration` from our public-facing docs. I don't know of any existing tests that validate that behavior -- do you @howardjohn ?  Our `Stackdriver Integration` can probably be promoted to Stable at this point.

douglas-reid commented on content/en/about/feature-stages/index.md:
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
 | [Service Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
 | [Istio Component Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
 | [Distributed Tracing](/docs/tasks/observability/distributed-tracing/) | Stable
-| [Stackdriver Integration](/docs/reference/config/policy-and-telemetry/adapters/stackdriver/) | Beta
```
this should be kept, but maybe file a bug to provide better docs?

ericvn replied:
The referenced docs, mixer/adapter/stackdriver/config/adapter.stackdriver.config.pb.html, were moved in the big removal PR, https://github.com/istio/istio/pull/25669. https://github.com/istio/istio/issues/26993 was created for a new doc to reference.

ericvn replied:
Also made `Stable`

douglas-reid commented on content/en/about/feature-stages/index.md:
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
 | [Distributed Tracing](/docs/tasks/observability/distributed-tracing/) | Stable
-| [Stackdriver Integration](/docs/reference/config/policy-and-telemetry/adapters/stackdriver/) | Beta
 | [Distributed Tracing to Zipkin / Jaeger](/docs/tasks/observability/distributed-tracing/) | Beta
 | [Logging with Fluentd](/docs/tasks/observability/mixer/logs/fluentd/) | Beta
```
this should be removed.

ericvn replied:
Leaving removed per second comment.

ericvn replied:
removed

douglas-reid commented on content/en/about/feature-stages/index.md:
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
 |-------------------|-------------------
 | [Prometheus Integration](/docs/tasks/observability/metrics/querying-metrics/) | Stable
 | [Local Logging (STDIO)](/docs/tasks/observability/mixer/logs/collecting-logs/) | Stable
```
this can be removed now

ericvn replied:
Removed in an earlier PR. Picked up removal in a rebase.

douglas-reid commented on content/en/about/feature-stages/index.md:
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
-| [Statsd Integration](/docs/reference/config/policy-and-telemetry/adapters/statsd/) | Stable
 | [Client and Server Telemetry Reporting](/docs/reference/config/policy-and-telemetry/) | Stable
 | [Service Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
 | [Istio Component Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
```
Maybe remove? This links to the same docs, and the dashboards have shifted a bit with the removal of all components except for Pilot.

ericvn replied:
removed

douglas-reid commented on scripts/grab_reference_docs.sh:
```diff
@@ -34,13 +34,8 @@ REPOS=(
-    https://github.com/osswangxining/alicloud-istio-grpcadapter.git@master
-    https://github.com/vmware/wavefront-adapter-for-istio.git@master
-    https://github.com/layer5io/layer5-istio-adapter.git@master
     https://github.com/apache/skywalking-data-collect-protocol.git@master
     https://github.com/ibm-cloud-security/app-identity-and-access-adapter.git@master
```
why not remove these as well?

ericvn replied:
Removed

douglas-reid commented on content/en/docs/reference/config/security/authorization-policy/index.html:
```diff
@@ -24,9 +24,19 @@
+<p>A request will be internally marked that it should be audited if there is an AUDIT policy on the workload that matches the request.
+A separate plugin must be configured and enabled to actually fulfill the audit decision and complete the audit behavior.
+The request will not be audited if there are no such supporting plugins enabled.
+Currently, the only supported plugin is the <a href="/docs/reference/config/proxy_extensions/stackdriver/">Telemetry v2 Stackdriver</a> plugin.</p>
```
I'm not sure where this came from but I'd like to remove the `Telemetry v2` before `Stackdriver`

ericvn replied:
This comes from istio.io/api. https://github.com/istio/api/pull/1618 created to remove.
